web/src/app/main/components/ToolbarAction.tsx

```diff
@@ -1,24 +1,29 @@
 import React from 'react'
 import Hidden from '@mui/material/Hidden'
+import { Theme } from '@mui/material/styles'
 import IconButton from '@mui/material/IconButton'
 import { Menu as MenuIcon, ChevronLeft } from '@mui/icons-material'
 import { useIsWidthDown } from '../../util/useWidth'
-import { PropTypes as p } from 'prop-types'
 import { Route, Switch, useLocation } from 'wouter'
 import { routes } from '../AppRoutes'
 
-function removeLastPartOfPath(path) {
+interface ToolbarActionProps {
+  showMobileSidebar: boolean
+  openMobileSidebar: (arg: boolean) => void
+}
+
+function removeLastPartOfPath(path: string): string {
   const parts = path.split('/')
   parts.pop()
   return parts.join('/')
 }
 
-function ToolbarAction(props) {
+function ToolbarAction(props: ToolbarActionProps): JSX.Element | null {
```

Comment:
Nitpick: return type could just be `JSX.Element`
For the instance when null was returned previously, we could just return a `<React.Fragment />` which is just an empty JSX element